src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -693,13 +712,10 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
         return daemon_spec
 
     def get_keyring(self, rgw_id: str) -> str:
-        ret, keyring, err = self.mgr.check_mon_command({
-            'prefix': 'auth get-or-create',
-            'entity': self.get_auth_entity(rgw_id),
-            'caps': ['mon', 'allow *',
-                     'mgr', 'allow rw',
-                     'osd', 'allow rwx tag rgw *=*'],
-        })
+        keyring = self.get_keyring_with_caps(self.get_auth_entity(rgw_id),
+                                             ['mon', 'allow *',
+                                              'mgr', 'allow rw',
+                                              'osd', 'allow rwx tag rgw *=*'])
```

Comment:
Would it be cleaner to just get-or-create (without any caps) and then set the caps afterwards?

Reply:
That would work but I sort of prefer to not have to set the caps every time. Given how infrequently they actually need to be updated (mostly just certain daemons in certain upgrades) it feels unnecessary.

Reply:
simply updating the caps after the fact might be simpler code-wise than doing the compare...?

Reply:
That's definitely true, the comparison is pretty messy because the inputs come as a list and come back from the command as a dict. Could be worth the extra commands to keep it simple. I'll make this change.